client/extensions/woocommerce/state/wc-api/reducer.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Internal dependencies
+ */
+import error from './error-reducer';
+import {
+	SERIALIZE,
+	DESERIALIZE,
+} from 'state/action-types';
+
+const initialState = {};
+
+const handlers = {
+	...error,
+};
+
+export default function( state = initialState, action ) {
+	const { type, payload } = action;
+	const { siteId } = payload || {};
+	const handler = handlers[ type ];
+
+	if ( handler ) {
+		// Necessary to ensure this data is not persisted.
+		if ( type === SERIALIZE || type === DESERIALIZE ) {
```

Comment:
I _think_ that the SERIALIZE/DESERIALIZE logic cannot fire unless the `handlers` defined here (or in a subsequent PR) includes a reducer that handles actions of one of those two types.

If this reducer is combined with others AND an action of type SERIALIZE or DESERIALIZE is passed to it, I _think_ your logic will only return `initialState` if you eventually add a handler for that action type.

This, I _think_ you might want to make this reducer safer to combine with others by moving the type check out of the handler check, unless I am mistaken.